daemon/containerd/image.go

```diff
@@ -137,6 +137,10 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 			}
 			refs = append(refs, name)
 
+			if _, nameWithDigest := name.(reference.Digested); nameWithDigest {
+				continue
+			}
```

Comment:
Perhaps a comment would be good here. Might use `ok` for the boolean, which is used most of the time, and helps cognitive load when "glancing" over the logic;

```suggestion
			if _, ok := name.(reference.Digested); ok {
				// Image name is  already a "digest", so no need to append a digest reference for it.
				continue
			}
```